yb-voyager/src/dbzm/schemaRegistry.go

```diff
@@ -40,10 +40,14 @@ type TableSchema struct {
 	Columns []Column `json:"columns"`
 }
 
-func (ts *TableSchema) getColumnType(columnName string) (string, error) {
+func (ts *TableSchema) getColumnType(columnName string, targetDBType string) (string, error) {
 	for _, colSchema := range ts.Columns {
 		if colSchema.Name == columnName {
-			if colSchema.Schema.Name != "" { // in case Kafka speicfic type which start with io.debezium...
+			if dbColType, ok := colSchema.Schema.Parameters["__debezium.source.column.type"]; ok && targetDBType == "oracle" &&
+				(strings.Contains(dbColType, "DATE") || strings.Contains(dbColType, "INTERVAL")) {
```

Comment:
What about for CHAR, TIMESTAMP, LONG? You've also included those types int he debezium config.

Reply:
there are different usage of all of them. See below else-if conditions. for example for CHAR types I am using length to set in the CTL file.

Reply:
ok, just wanted to ensure you didn't miss it